lib/manager/npm/post-update/yarn.spec.ts

```diff
@@ -97,6 +97,27 @@ describe('manager/npm/post-update/yarn', () => {
     expect(fixSnapshots(execSnapshots)).toMatchSnapshot();
   });
 
+  it('does not use global cache if zero install is detected', async () => {
+    const execSnapshots = mockExecAll(exec, {
+      stdout: '2.1.0',
+      stderr: '',
+    });
+    fs.readFile.mockImplementation(
+      (filename, encoding) =>
+        new Promise<string>((resolve) => resolve('package-lock-contents'))
```

Comment:
```suggestion
        Promise.resolve('package-lock-contents')
```

Simplify code. I'm sure `fs.readFile.mockResolvedValueOnce('package-lock-contents')` should work too.

Please use once mocking instead of general mock: `mockImplementationOnce`